### PR TITLE
Create new project with the user interface

### PR DESCRIPTION
**Pull request by masolano**
Create new project of LocalMatch of the user interface with the login interface, signup interface and main interface.

### CODE REVIEW

File: localMatchApp/mainwindow.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#include <QMessageBox>
+
+MainWindow::MainWindow(QWidget *parent) :
+    QMainWindow(parent),
+    ui(new Ui::MainWindow)
+{
+    ui->setupUi(this);
+    QPixmap pix ("/home/marcelino/Desktop/LocalMatch/images/logo.png");
+    ui->pictureLogo->setPixmap(pix);
+    this->setFixedWidth(500);
```
Comment (luvalverde) [inline localMatchApp/mainwindow.cpp]: do not use magic numbers

File: localMatchApp/mainwindow.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+#include "homewindow.h"
+#include "QDesktopWidget"
+#include <QMessageBox>
+
+MainWindow::MainWindow(QWidget *parent) :
+    QMainWindow(parent),
+    ui(new Ui::MainWindow)
+{
+    ui->setupUi(this);
+    QPixmap pix ("/home/marcelino/Desktop/LocalMatch/images/logo.png");
```
Comment (luvalverde) [inline localMatchApp/mainwindow.cpp]: do not use hardcoded paths

File: localMatchApp/mainwindow.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        )
+    );
+}
+
+MainWindow::~MainWindow()
+{
+    delete ui;
+}
+
+void MainWindow::on_buttonSignUp_clicked()
```
Comment (luvalverde) [inline localMatchApp/mainwindow.cpp]: According with the coding standard, underscores must never be used in an identifier.
Please apply this comment in all your files

File: localMatchApp/mainwindow.cpp [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+{
+    SignUp signButton;
+    signButton.setModal(true);
+    signButton.exec();
+}
+
+void MainWindow::on_buttonLogin_clicked()
+{
+    //if(ui->lineEditUser->text()=="marcelino" && ui->lineEditPassword->text()=="123"){
```
Comment (luvalverde) [inline localMatchApp/mainwindow.cpp]: let's avoid to merge into master code that is commented or not being used

File: localMatchApp/signup.cpp [hunk 5]
```diff
@@ -0,0 +1,39 @@
+#include "signup.h"
+#include "ui_signup.h"
+#include "QDesktopWidget"
+#include <QMessageBox>
+
+SignUp::SignUp(QWidget *parent) :
```
Comment (luvalverde) [inline localMatchApp/signup.cpp]: according with the coding standard, function identifiers must have identifier names that begin with lowercase letters. Also, arguments whose input value is used by the function, and whose value is not changed by the function should be prefixed by "i_" (i.e. input).  

Please apply this comment in all your files.

File: localMatchApp/signup.cpp [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    QDialog(parent),
+    ui(new Ui::SignUp)
+{
+    ui->setupUi(this);
+    ui->lineEdit_NewUser->setPlaceholderText("Username");
+    ui->lineEdit_NewPassword->setPlaceholderText("Password");
+    ui->lineEdit_NewPassCheck->setPlaceholderText("Repeat Password");
+    QPixmap pix ("/home/marcelino/Desktop/LocalMatch/images/logo.png");
+    ui->picture->setPixmap(pix);
+    this->setFixedWidth(500);
```
Comment (luvalverde) [inline localMatchApp/signup.cpp]: avoid magical numbers

File: localMatchApp/signup.cpp [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+SignUp::SignUp(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::SignUp)
+{
+    ui->setupUi(this);
+    ui->lineEdit_NewUser->setPlaceholderText("Username");
+    ui->lineEdit_NewPassword->setPlaceholderText("Password");
+    ui->lineEdit_NewPassCheck->setPlaceholderText("Repeat Password");
+    QPixmap pix ("/home/marcelino/Desktop/LocalMatch/images/logo.png");
```
Comment (luvalverde) [inline localMatchApp/signup.cpp]: let's use relative paths

**Review by luvalverde — CHANGES_REQUESTED** [review]
I added some comments.
Also, let's use vectored images instead of .png or jpg.

https://vimm.com/what-is-a-vector-image/

Also:
- the image name "event1.jpg" is not very representative
- Why are there 2 same images ("logo.png") just differing in their size? The image must adjust to its corresponding container size.

**Comment by luvalverde** [comment]
Like this will be the base of the project, would be useful to define directories to separate:

header files -> headers
source code -> src
tests ->test
so, the following code will be located in an standard place

File: localMatchApp/mainwindow.cpp (hunk 3 above)
Comment (luvalverde) [reply localMatchApp/mainwindow.cpp]: Check out this link:
http://doc.qt.io/qt-5/designer-using-a-ui-file.html#a-dialog-without-auto-connect

File: localMatchApp/signup.cpp [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+SignUp::~SignUp()
+{
+    delete ui;
+}
+
+void SignUp::on_NewUserButton_clicked()
+{
+    QMessageBox* msgbox = new QMessageBox(this);
```
Comment (wfallascr) [inline localMatchApp/signup.cpp]: Memory leak. Please, contemplate the option of using smart pointers here or any other RAII approach.

**Review by wfallascr — CHANGES_REQUESTED** [review]

File: localMatchApp/mainwindow.cpp (hunk 1 above)
Comment (masolano) [reply localMatchApp/mainwindow.cpp]: Done

File: localMatchApp/mainwindow.cpp (hunk 2 above)
Comment (masolano) [reply localMatchApp/mainwindow.cpp]: @Done

File: localMatchApp/headers/homewindow.h [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+namespace Ui {
+class HomeWindow;
+}
+
+class HomeWindow : public QDialog
+{
+    Q_OBJECT
+
+public:
+    explicit HomeWindow(QWidget *parent = nullptr);
```
Comment (luvalverde) [inline localMatchApp/headers/homewindow.h]: i_parent

File: localMatchApp/headers/mainwindow.h [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+namespace Ui {
+class MainWindow;
+}
+
+class MainWindow : public QMainWindow
+{
+    Q_OBJECT
+
+public:
+    explicit MainWindow(QWidget *parent = 0);
```
Comment (luvalverde) [inline localMatchApp/headers/mainwindow.h]: i_parent

File: localMatchApp/sources/homewindow.cpp [hunk 11]
```diff
@@ -0,0 +1,32 @@
+#include "../headers/homewindow.h"
+#include "ui_homewindow.h"
+
+HomeWindow::HomeWindow(QWidget *i_parent) :
+    QDialog(i_parent),
+    ui(new Ui::HomeWindow)
+{
+    int widthSizeLogo = 150;
```
Comment (luvalverde) [inline localMatchApp/sources/homewindow.cpp]: check the coding standard for Magical Numbers

File: localMatchApp/sources/mainwindow.cpp [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+#include "../headers/signup.h"
+#include "../headers/homewindow.h"
+#include "QDesktopWidget"
+#include <QMessageBox>
+
+MainWindow::MainWindow(QWidget *i_parent) :
+    QMainWindow(i_parent),
+    ui(new Ui::MainWindow)
+{
+    int fixedWidthMainWindow = 500;
```
Comment (luvalverde) [inline localMatchApp/sources/mainwindow.cpp]: check the coding standard for Magical Numbers

File: localMatchApp/sources/mainwindow.cpp [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+void MainWindow::onButtonSignUpClicked()
+{
+    SignUp signUpWindow;
+    signUpWindow.setModal(true);
+    signUpWindow.exec();
+}
+
+void MainWindow::onButtonLoginClicked()
+{
+    if(ui->lineEditUser->text()=="masolano" && ui->lineEditPassword->text()=="123"){
```
Comment (luvalverde) [inline localMatchApp/sources/mainwindow.cpp]: Add a comment for these hardcoded values.
Add a TODO comment indicating this must be changed.

File: localMatchApp/sources/signup.cpp [hunk 14]
```diff
@@ -0,0 +1,42 @@
+#include "../headers/signup.h"
+#include "ui_signup.h"
+#include "QDesktopWidget"
+#include <QMessageBox>
+
+SignUp::SignUp(QWidget *i_parent) :
+    QDialog(i_parent),
+    ui(new Ui::SignUp)
+{
+    int fixedWidthSignUpWindow = 500;
```
Comment (luvalverde) [inline localMatchApp/sources/signup.cpp]: check the coding standard for Magical Numbers

**Review by luvalverde — CHANGES_REQUESTED** [review]
Some comments

File: localMatchApp/LocalMatch.pro.user [hunk 15]
```diff
@@ -0,0 +1,338 @@
...
+  <variable>ProjectExplorer.Project.Target.0</variable>
+  <valuemap type="QVariantMap">
+   <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Desktop</value>
+   <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName">Desktop</value>
+   <value type="QString" key="ProjectExplorer.ProjectConfiguration.Id">{19164225-e683-4d8b-b487-b302a6b7d164}</value>
+   <value type="int" key="ProjectExplorer.Target.ActiveBuildConfiguration">0</value>
+   <value type="int" key="ProjectExplorer.Target.ActiveDeployConfiguration">0</value>
+   <value type="int" key="ProjectExplorer.Target.ActiveRunConfiguration">0</value>
+   <valuemap type="QVariantMap" key="ProjectExplorer.Target.BuildConfiguration.0">
+    <value type="QString" key="ProjectExplorer.BuildConfiguration.BuildDirectory">/home/marcelino/Desktop/build-LocalMatch-Desktop-Debug</value>
```
Comment (luvalverde) [inline localMatchApp/LocalMatch.pro.user]: this seems to be hard-coded to @masolano computer
Let's take a look at this
http://doc.qt.io/qtcreator/creator-sharing-project-settings.html

File: localMatchApp/LocalMatch.pro.user.3.2-pre1 [hunk 16]
```diff
@@ -0,0 +1,260 @@
...
+  <variable>ProjectExplorer.Project.Target.0</variable>
+  <valuemap type="QVariantMap">
+   <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Desktop</value>
+   <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName">Desktop</value>
+   <value type="QString" key="ProjectExplorer.ProjectConfiguration.Id">{f6d61700-0030-43d9-8e5b-b08c7cab44c3}</value>
+   <value type="int" key="ProjectExplorer.Target.ActiveBuildConfiguration">0</value>
+   <value type="int" key="ProjectExplorer.Target.ActiveDeployConfiguration">0</value>
+   <value type="int" key="ProjectExplorer.Target.ActiveRunConfiguration">0</value>
+   <valuemap type="QVariantMap" key="ProjectExplorer.Target.BuildConfiguration.0">
+    <value type="QString" key="ProjectExplorer.BuildConfiguration.BuildDirectory">/home/andres/build-LocalMatch-Desktop-Debug</value>
```
Comment (luvalverde) [inline localMatchApp/LocalMatch.pro.user.3.2-pre1]: this seems to be hard-coded to @anpineda computer

File: localMatchApp/sources/homewindow.cpp [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+    ui->pictureLogo->setPixmap(pixLogo);
+    ui->labelNameUser->setText("Marcelino Solano");
+    ui->labelNameEvent->setText("New Year Party 2018");
+    ui->labelNameEvent->setStyleSheet("font-weight: bold;");
+    ui->labelDateEvent->setText("05/12/2018");
+    ui->labelDateEvent->setStyleSheet("font-weight: bold; color: rgb(187, 205, 225);");
+    ui->labelHourEvent->setText("18:00 - 23:00");
+    ui->labelHourEvent->setStyleSheet("font-weight: bold; color: rgb(187, 205, 225);");
+    ui->labelDescription->setWordWrap(true);
+    ui->labelDescription->setText("So perhaps, you've generated some fancy text, and you're content that you can now copy and paste your fancy text in the comments section of funny cat videos, but perhaps you're wondering how it's even possible to change the font of your text? Is it some sort of hack? Are you copying and pasting an actual font?");
```
Comment (luvalverde) [inline localMatchApp/sources/homewindow.cpp]: according with the coding standard, the maximum recommended length of a line of code including white-space is 80 characters.

**Review by luvalverde — CHANGES_REQUESTED** [review]
I added some comments

**Comment by masolano** [comment]
PR back to you

File: localMatchApp/sources/homewindow.cpp [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+    ui->setupUi(this);
+    QPixmap pixEvent (":/images/eventExample.svg");
+    QPixmap pixUser (":/images/user.svg");
+    QPixmap pixLogo (":/images/logo.svg");
+    pixLogo = pixLogo.scaled(WIDTHSIZELOGO, HEIGHTSIZELOGO, Qt::KeepAspectRatio,
+                             Qt::SmoothTransformation);
+    ui->pictureEvent->setPixmap(pixEvent);
+    ui->pictureUser->setPixmap(pixUser);
+    ui->pictureLogo->setPixmap(pixLogo);
+    ui->labelNameUser->setText("Marcelino Solano");
```
Comment (wfallascr) [inline localMatchApp/sources/homewindow.cpp]: Why this value?

File: localMatchApp/headers/signup.h [hunk 19]
```diff
@@ -0,0 +1,25 @@
+#ifndef SIGNUP_H
+#define SIGNUP_H
+
+#include <QDialog>
+
+namespace Ui {
+class SignUp;
+}
```
Comment (wfallascr) [inline localMatchApp/headers/signup.h:12]: Please, add doxygen documentation of all your classes/functions

**Review by wfallascr — CHANGES_REQUESTED** [review]

**Comment by masolano** [comment]
PR back to you

File: localMatchApp/sources/homewindow.cpp [hunk 20]
```diff
@@ -0,0 +1,51 @@
+#include "../headers/homewindow.h"
+#include "ui_homewindow.h"
+#include <QStyle>
+
+/*!
+ * In this function its created what is necessary
```
Comment (luvalverde) [inline localMatchApp/sources/homewindow.cpp]: In this function is created ...

File: localMatchApp/sources/mainwindow.cpp [hunk 21]
```diff
@@ -0,0 +1,70 @@
+#include "../headers/mainwindow.h"
+#include "ui_mainwindow.h"
+#include "../headers/signup.h"
+#include "../headers/homewindow.h"
+#include "QDesktopWidget"
+#include <QMessageBox>
+#include <QStyle>
+
+/*!
+ * In this function its created what is necessary to
```
Comment (luvalverde) [inline localMatchApp/sources/mainwindow.cpp]: In this function is created ...

File: localMatchApp/sources/signup.cpp [hunk 22]
```diff
@@ -0,0 +1,56 @@
+#include "../headers/signup.h"
+#include "ui_signup.h"
+#include "QDesktopWidget"
+#include <QMessageBox>
+#include <QStyle>
+
+/*!
+ * In this function its created what is necessary to build
```
Comment (luvalverde) [inline localMatchApp/sources/signup.cpp]: In this function is created ...

**Review by luvalverde — CHANGES_REQUESTED** [review]
Just minor comments.
Please squash your commits

**Comment by masolano** [comment]
PR back to you

**Review by wfallascr — APPROVED** [review]

**Review by luvalverde — APPROVED** [review]
Nice

**Pull request merged**